NullPointerException in JSON Deserializer

**Describe the bug**
Our jenkins pipeline worked well, but then I deleted a long living branch in sonarqube (not in git!) to be able to rename the default branch from master to the name of the long living branch. Then the pipeline stopped to work.

**Expected behavior**
The gradle plugin should run even if there is no existing analysis.

**Stacktrace**
```
Caused by: java.lang.NullPointerException
        at com.github.mc1arke.sonarqube.plugin.scanner.CommunityProjectPullRequestsLoader.lambda$createPullRequestInfoJsonDeserialiser$0(CommunityProjectPullRequestsLoader.java:76)
        at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69)
        at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41)
        at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82)
        at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
        at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
        at com.google.gson.Gson.fromJson(Gson.java:922)
        at com.google.gson.Gson.fromJson(Gson.java:860)
        at com.github.mc1arke.sonarqube.plugin.scanner.CommunityProjectPullRequestsLoader.load(CommunityProjectPullRequestsLoader.java:87)
        at org.sonar.scanner.scan.branch.ProjectPullRequestsProvider.provide(ProjectPullRequestsProvider.java:47)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.picocontainer.injectors.MethodInjector.invokeMethod(MethodInjector.java:129)
        at org.picocontainer.injectors.MethodInjector.access$000(MethodInjector.java:39)
        at org.picocontainer.injectors.MethodInjector$2.run(MethodInjector.java:113)
        at org.picocontainer.injectors.AbstractInjector$ThreadLocalCyclicDependencyGuard.observe(AbstractInjector.java:270)
        at org.picocontainer.injectors.MethodInjector.decorateComponentInstance(MethodInjector.java:120)
        at org.picocontainer.injectors.CompositeInjector.decorateComponentInstance(CompositeInjector.java:58)
        at org.picocontainer.injectors.Reinjector.reinject(Reinjector.java:142)
        at org.picocontainer.injectors.ProviderAdapter.getComponentInstance(ProviderAdapter.java:96)
        at org.picocontainer.DefaultPicoContainer.getInstance(DefaultPicoContainer.java:699)
        at org.picocontainer.DefaultPicoContainer.getComponent(DefaultPicoContainer.java:647)
        at org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer.getComponent(ComponentContainer.java:63)
        ... 150 more
```

**Software Versions**
 - SonarQube Version: 7.8
 - Plugin Version: version contained in sonar gradle plugin v2.7.1 and v2.8

--- a/src/main/java/com/github/mc1arke/sonarqube/plugin/scanner/CommunityProjectPullRequestsLoader.java
+++ b/src/main/java/com/github/mc1arke/sonarqube/plugin/scanner/CommunityProjectPullRequestsLoader.java
@@ -41,6 +41,7 @@ import java.text.ParseException;
 import java.text.SimpleDateFormat;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
 /**
  * Loads the Pull Requests currently known by SonarQube from the server component for client applications.
@@ -73,8 +74,7 @@ public class CommunityProjectPullRequestsLoader implements ProjectPullRequestsLo
             } catch (ParseException e) {
                 LOGGER.warn("Could not parse date from Pull Requests API response. Will use '0' date", e);
             }
-            final JsonElement baseJson = jsonObject.get("base");
-            final String base = baseJson != null ? baseJson.getAsString() : null;
+            final String base = Optional.ofNullable(jsonObject.get("base")).map(JsonElement::getAsString).orElse(null);
             return new PullRequestInfo(jsonObject.get("key").getAsString(), jsonObject.get("branch").getAsString(),
                     base, parsedDate);
         };

--- a/src/main/java/com/github/mc1arke/sonarqube/plugin/scanner/CommunityProjectPullRequestsLoader.java
+++ b/src/main/java/com/github/mc1arke/sonarqube/plugin/scanner/CommunityProjectPullRequestsLoader.java
@@ -21,6 +21,7 @@ package com.github.mc1arke.sonarqube.plugin.scanner;
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
 import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
 import org.sonar.api.utils.MessageException;
 import org.sonar.api.utils.log.Logger;
@@ -72,8 +73,10 @@ public class CommunityProjectPullRequestsLoader implements ProjectPullRequestsLo
             } catch (ParseException e) {
                 LOGGER.warn("Could not parse date from Pull Requests API response. Will use '0' date", e);
             }
+            final JsonElement baseJson = jsonObject.get("base");
+            final String base = baseJson != null ? baseJson.getAsString() : null;
             return new PullRequestInfo(jsonObject.get("key").getAsString(), jsonObject.get("branch").getAsString(),
-                                       jsonObject.get("base").getAsString(), parsedDate);
+                    base, parsedDate);
         };
     }
 

--- a/src/test/java/com/github/mc1arke/sonarqube/plugin/scanner/CommunityProjectPullRequestsLoaderTest.java
+++ b/src/test/java/com/github/mc1arke/sonarqube/plugin/scanner/CommunityProjectPullRequestsLoaderTest.java
@@ -40,6 +40,7 @@ import java.util.ArrayList;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.mock;
@@ -112,6 +113,27 @@ public class CommunityProjectPullRequestsLoaderTest {
         assertNotNull(responseInfo);
         assertEquals(0, responseInfo.getAnalysisDate());
         assertEquals("master", responseInfo.getBase());
+        assertEquals("dummybranch", responseInfo.getBranch());
+        assertEquals("101", responseInfo.getKey());
+    }
+
+    @Test
+    public void testAllPullRequestsFromNonEmptyServerResponseWithoutBase() {
+        WsResponse mockResponse = mock(WsResponse.class);
+        when(scannerWsClient.call(any())).thenReturn(mockResponse);
+
+        StringReader stringReader = new StringReader(
+                "{\"pullRequests\":[{\"key\":\"101\",\"title\":\"dummybranch\",\"branch\":\"dummybranch\",\"status\":{\"qualityGateStatus\":\"OK\",\"bugs\":0,\"vulnerabilities\":0,\"codeSmells\":0},\"analysisDate\":\"\"}]}");
+        when(mockResponse.contentReader()).thenReturn(stringReader);
+
+        CommunityProjectPullRequestsLoader testCase = new CommunityProjectPullRequestsLoader(scannerWsClient);
+        ProjectPullRequests response = testCase.load("key");
+        assertFalse(response.isEmpty());
+
+        PullRequestInfo responseInfo = response.get("dummybranch");
+        assertNotNull(responseInfo);
+        assertEquals(0, responseInfo.getAnalysisDate());
+        assertNull(responseInfo.getBase());
         assertEquals("dummybranch", responseInfo.getBranch());
         assertEquals("101", responseInfo.getKey());
     }

